crypto-ffi/src/wasm.rs

```diff
@@ -1584,10 +1583,187 @@ impl CoreCrypto {
             .err_into(),
         )
     }
+
+    /// see [core_crypto::MlsCentral::new_acme_enrollment]
+    pub async fn new_acme_enrollment(&self, ciphersuite: Ciphersuite) -> WasmCryptoResult<WireE2eIdentity> {
+        let ciphersuite: CiphersuiteName = ciphersuite.into();
+        let enrollment = self
+            .0
+            .read()
+            .await
+            .new_acme_enrollment(ciphersuite.into())
+            .map(WireE2eIdentity)
+            .map_err(|_| WasmCryptoError::new("Implementation error"))?;
+        WasmCryptoResult::Ok(enrollment)
+    }
 }
 
 #[wasm_bindgen]
 /// Returns the current version of CoreCrypto
 pub fn version() -> String {
     crate::VERSION.into()
 }
+
+#[derive(Debug)]
+#[wasm_bindgen]
+#[repr(transparent)]
+pub struct WireE2eIdentity(core_crypto::prelude::WireE2eIdentity);
+
+#[wasm_bindgen]
+impl WireE2eIdentity {
+    pub fn directory_response(&self, directory: Vec<u8>) -> WasmCryptoResult<JsValue> {
+        let directory: AcmeDirectory = self.0.directory_response(directory)?.into();
+        WasmCryptoResult::Ok(serde_wasm_bindgen::to_value(&directory)?)
+    }
+
+    pub fn new_account_request(&self, directory: JsValue, previous_nonce: String) -> WasmCryptoResult<Uint8Array> {
+        let directory = serde_wasm_bindgen::from_value::<AcmeDirectory>(directory)?;
+        let new_account = self.0.new_account_request(directory.try_into()?, previous_nonce)?;
+        WasmCryptoResult::Ok(new_account.as_slice().into())
+    }
+
+    pub fn new_account_response(&self, account: Uint8Array) -> WasmCryptoResult<Uint8Array> {
+        let account: Vec<u8> = self.0.new_account_response(account.to_vec())?.into();
+        WasmCryptoResult::Ok(account.as_slice().into())
+    }
+
+    pub fn new_order_request(
+        &self,
+        handle: String,
+        client_id: String,
+        expiry_days: u32,
+        directory: JsValue,
+        account: Uint8Array,
+        previous_nonce: String,
+    ) -> WasmCryptoResult<Uint8Array> {
+        let directory = serde_wasm_bindgen::from_value::<AcmeDirectory>(directory)?;
+        let new_order = self.0.new_order_request(
+            handle,
+            client_id,
+            expiry_days,
+            directory.try_into()?,
+            account.to_vec().into(),
+            previous_nonce,
+        )?;
+        WasmCryptoResult::Ok(new_order.as_slice().into())
+    }
+
+    /*pub fn new_order_response(&self, order: Uint8Array) -> WasmCryptoResult<Uint8Array> {
+        let order: Vec<u8> = self.0.new_order_response(order.to_vec())?.into();
+        WasmCryptoResult::Ok(order.as_slice().into())
+    }*/
+}
+
+#[wasm_bindgen]
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
```

Comment:
Unsure but you might need to add something like `#[serde(rename_all = "camelCase")]` for the JSON conversion to happen correctly between the JS naming schemes and the rust-native `snake_case`